### PR TITLE
[FIX] ui5Framework: Improve error handling for duplicate lib declaration

### PR DESCRIPTION
**Pull request by matz3**
Fixes: https://github.com/SAP/ui5-tooling/issues/780

### CODE REVIEW

File: lib/graph/helpers/ui5Framework.js [hunk 1]
```diff
@@ -192,6 +192,22 @@ const utils = {
 		});
 		await projectGraph.resolveOptionalDependencies();
 	},
+	checkForDuplicateFrameworkProjects(projectGraph, frameworkGraph) {
+		// Check for duplicate framework libraries
+		const projectGraphProjectNames = projectGraph.getProjectNames();
+		const duplicateFrameworkProjectNames = frameworkGraph.getProjectNames()
+			.filter((name) => projectGraphProjectNames.includes(name));
+
+		if (duplicateFrameworkProjectNames.length) {
```
Comment (d3xter666) [inline lib/graph/helpers/ui5Framework.js:201]: Maybe `Array.some()` would suit better here:

```suggestion
			.some((name) => projectGraphProjectNames.includes(name));

		if (duplicateFrameworkProjectNames) {
```

**Review by d3xter666 — COMMENTED** [review]
LGTM

File: lib/graph/helpers/ui5Framework.js [hunk 2]
```diff
@@ -192,6 +192,22 @@ const utils = {
...
+			throw new Error(
+				`Duplicate framework library definition(s) found in project ${projectGraph.getRoot().getName()}: ` +
```
Comment (RandomByte) [inline lib/graph/helpers/ui5Framework.js:203]:
```suggestion
				`Duplicate framework dependency definition(s) found for project ${projectGraph.getRoot().getName()}: ` +
```

File: lib/graph/helpers/ui5Framework.js [hunk 3]
```diff
@@ -192,6 +192,22 @@ const utils = {
...
+				`${duplicateFrameworkProjectNames.join(", ")}. ` +
```
Comment (RandomByte) [inline lib/graph/helpers/ui5Framework.js:204]:
```suggestion
				`${duplicateFrameworkProjectNames.join(", ")}.\n` +
```
Line break for readability?

File: lib/graph/helpers/ui5Framework.js [hunk 4]
```diff
@@ -192,6 +192,22 @@ const utils = {
...
+				`Framework libraries should only be referenced via ui5.yaml configuration, ` +
+				`not in its dependencies (e.g. package.json). ` +
+				`Note that this error could also come from transitive dependencies.`
```
Comment (RandomByte) [inline lib/graph/helpers/ui5Framework.js:207]:
```suggestion
				`Framework libraries should only be referenced via ui5.yaml configuration. ` +
				`Neither the root project, nor any of its dependencies should include them as direct ` +
				`dependencies (e.g. via package.json).`
```

What do you think? I'm not sure how common the term "transitive dependencies" is.

**Review by RandomByte — CHANGES_REQUESTED** [review]

**Review by RandomByte — APPROVED** [review]

**Pull request merged**